### ISSUE TITLE
opentracing middleware: Add Option to put proto request in span for certain methods.

### ISSUE THREAD

**Issue by bwplotka** [grpc-ecosystem/go-grpc-middleware#273]
AC:
* We can enrich tracing in a generic way.

**Comment by jackskj** [comment]
If this issue is not in development, I can take a stab at it, it would come in handy.

**Comment by bwplotka** [comment]
Yes please help! Make sure you to actually do it on `v2` branch which will be released soon. See: https://github.com/grpc-ecosystem/go-grpc-middleware/issues/275

Also I think we might need help in https://github.com/grpc-ecosystem/go-grpc-middleware/issues/277 which might more interesting to solve (:

**Comment by jackskj** [comment]
Oh I see, #277 seems like a it should be resolved before this issue. 
I’ll take a stab at it,

**Comment by jackskj** [comment]
However, pushing entire request in the spawn may cause unnecessarily large amount of data collection. Do you think using field option is a worthy addition?
`
syntax = "proto3";
import "google/protobuf/descriptor.proto";
package tracing;
extend google.protobuf.FieldOptions {
    bool trace = 50000;
}

message MyMessage {
    string stuff = 1 [(tracing.trace) = true];
}
`

**Comment by bwplotka** [comment]
Well it depends on your method. Depends how big requests you have. (or
responses). I would love to have this for certain methods (:
